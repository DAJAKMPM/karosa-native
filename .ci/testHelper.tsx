import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
