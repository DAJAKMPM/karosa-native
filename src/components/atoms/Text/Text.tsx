/**
 *
 * Text
 * @format
 *
 */

import React, { FC } from "react";
import { Text as RnText } from "react-native-elements";

import type { PropsType } from "./types";
import TextStyles from "./styles";

const Text: FC<PropsType> = (props) => {
  const { text, customStyle, numberOfLines, ellipsizeMode } = props;

  return (
    <RnText
      style={[TextStyles.text, customStyle]}
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
    >
      {text}
    </RnText>
  );
};

export default Text;
