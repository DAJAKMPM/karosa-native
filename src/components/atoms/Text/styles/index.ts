/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const TextStyles = StyleSheet.create({
  text: {
    ...theme.textRegular,
    color: theme.colors.dark20,
    fontWeight: "400",
  },
});

export default TextStyles;
