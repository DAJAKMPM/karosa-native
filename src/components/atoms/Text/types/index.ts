/**
 *
 * @format
 *
 */

import { StyleProp, TextStyle } from "react-native";

type EllipsizeModeType = "head" | "middle" | "tail" | "clip";

export type PropsType = {
  text: string;
  customStyle?: StyleProp<TextStyle>;
  numberOfLines?: number;
  ellipsizeMode?: EllipsizeModeType;
};
