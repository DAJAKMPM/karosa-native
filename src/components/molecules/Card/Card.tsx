/**
 *
 * Card
 * @format
 *
 */

import React, { FC } from "react";
import { TouchableWithoutFeedback } from "react-native";
import { Card as RnCard } from "react-native-elements";

import type { PropsType } from "./types";

const Card: FC<PropsType> = (props) => {
  const { children, containerStyle, wrapperStyle, onPress } = props;

  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <RnCard containerStyle={containerStyle} wrapperStyle={wrapperStyle}>
        {children}
      </RnCard>
    </TouchableWithoutFeedback>
  );
};

export default Card;
