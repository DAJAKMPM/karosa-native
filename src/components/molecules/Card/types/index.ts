/**
 *
 * @format
 *
 */

import { StyleProp, ViewStyle } from "react-native";

export type PropsType = {
  containerStyle?: StyleProp<ViewStyle>;
  wrapperStyle?: StyleProp<ViewStyle>;
};
