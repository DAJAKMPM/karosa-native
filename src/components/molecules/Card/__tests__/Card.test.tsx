import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Text from "@app/atoms/Text";
import Card from "../Card";

describe("Card Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Card>
        <Text text="test" />
      </Card>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
