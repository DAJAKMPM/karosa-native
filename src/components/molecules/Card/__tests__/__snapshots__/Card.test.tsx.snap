// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 18.75,
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
